firebase-vscode/src/data-connect/ad-hoc-mutations.ts

```diff
@@ -162,46 +176,79 @@ query {
     }
   }
 
-  async function generateMutation(
-    ast: ObjectTypeDefinitionNode,
-  ): Promise<string> {
-    const introspect = (await dataConnectService.introspect())?.data;
-    const schema = buildClientSchema(introspect!);
+  function makeAdHocMutation(
+    fields: GraphQLInputField[],
+    singularName: string,
+  ): OperationDefinitionNode {
+    const argumentFields: ObjectFieldNode[] = [];
 
-    const name = ast.name.value;
-    const lowerCaseName =
-      ast.name.value.charAt(0).toLowerCase() + ast.name.value.slice(1);
-    const dataName = `${name}_Data`;
-    const mutationDataType: GraphQLInputObjectType = schema.getTypeMap()[
-      dataName
-    ] as GraphQLInputObjectType;
-
-    // build mutation as string
-    const functionSpacing = "\t";
-    const fieldSpacing = "\t\t";
-    const mutation = [];
-    mutation.push("mutation {"); // mutation header
-    mutation.push(`${functionSpacing}${lowerCaseName}_insert(data: {`);
-    for (const [fieldName, field] of Object.entries(
-      mutationDataType.getFields(),
-    )) {
-      // necessary to avoid type error
-      const fieldtype: any = field.type;
-      // use all argument types that are of scalar, except x_expr
-      if (
-        isDataConnectScalarType(fieldtype.name) &&
-        !field.name.includes("_expr")
-      ) {
-        const defaultValue = (defaultScalarValues as any)[fieldtype.name] || "";
-        mutation.push(
-          `${fieldSpacing}${fieldName}: ${defaultValue} # ${fieldtype.name}`,
-        ); // field name + temp value + comment
-      }
+    for (const field of fields) {
+      const type = getNamedType(field.type);
+      const defaultValue = getDefaultScalarValue(type.name);
+      if (!defaultValue) continue;
+
+      argumentFields.push({
+        kind: Kind.OBJECT_FIELD,
+        name: { kind: Kind.NAME, value: field.name },
+        value: defaultValue,
+      });
     }
-    mutation.push(`${functionSpacing}})`, "}"); // closing braces/paren
-    return mutation.join("\n");
+
+    return {
+      kind: Kind.OPERATION_DEFINITION,
+      operation: OperationTypeNode.MUTATION,
+      name: { kind: Kind.NAME, value: singularName },
```

Comment:
Let's keep it anonymous

```suggestion
```

Reply:
If it's anonymous, do we run into problems with folks pressing AddData multiple times? Or do we make a new file each time? I forget how the extension works here

Reply:
I check if there's a file first, so it won't make a bunch of files